iotdb-core/node-commons/src/main/java/org/apache/iotdb/commons/udf/UDFTable.java

```diff
@@ -19,99 +19,101 @@
 
 package org.apache.iotdb.commons.udf;
 
-import org.apache.iotdb.commons.udf.builtin.BuiltinTimeSeriesGeneratingFunction;
+import org.apache.iotdb.common.rpc.thrift.Model;
 import org.apache.iotdb.commons.udf.service.UDFClassLoader;
 import org.apache.iotdb.commons.utils.TestOnly;
 
+import org.apache.tsfile.utils.Pair;
 import org.apache.tsfile.utils.ReadWriteIOUtils;
 
 import java.io.IOException;
 import java.io.InputStream;
 import java.io.OutputStream;
+import java.util.ArrayList;
 import java.util.List;
 import java.util.Map;
 import java.util.concurrent.ConcurrentHashMap;
 import java.util.stream.Collectors;
 
+/**
+ * UDFTable is a table that stores UDF information. On DataNode, it stores all UDF information. On
+ * ConfigNode, it does not store built-in UDF information.
+ */
 public class UDFTable {
-  private final Map<String, UDFInformation> udfInformationMap;
 
-  /** maintain a map for creating instance */
-  private final Map<String, Class<?>> functionToClassMap;
+  /** functionName -> information * */
+  private final Map<Pair<Model, String>, UDFInformation> udfInformationMap;
+
+  /** maintain a map for creating instance, functionName -> class */
+  private final Map<Pair<Model, String>, Class<?>> functionToClassMap;
```

Comment:
Why not using Map<Model, Map<String, UDFInformation>>? In this way, you won't need to iterate all map entry while trying to all functions of one kind of model.

Reply:
Done. Use `Map<Model, Map<String, UDFInformation>>` now